packages/syft/src/syft/types/syft_object.py

```diff
@@ -44,14 +45,37 @@
 HIGHEST_SYFT_OBJECT_VERSION = max(supported_object_versions)
 LOWEST_SYFT_OBJECT_VERSION = min(supported_object_versions)
 
+DEFAULT_HASH_EXCLUDE_ATTRS = [
+    "syft_node_location",
+    "syft_client_verify_key",
+]
 
-class SyftBaseObject(BaseModel):
+
+class SyftHashableObject:
+    __hash_exclude_attrs__ = []
+
+    def __hash__(self) -> int:
+        return int.from_bytes(self.__sha256__(), byteorder="big")
+
+    def __sha256__(self) -> bytes:
+        self.__hash_exclude_attrs__.extend(DEFAULT_HASH_EXCLUDE_ATTRS)
```

Comment:
But the problem is that any subclass defining __hash_exclude_attrs__ will overwrite this. So we want a behaviour where we can always exclude `syft_node_location` and `syft_client_verify_key` in the hash comparison. Otherwise, I need to define both of these attributes in all subclasses where __hash_exclude_attrs__ is overwritten.